src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,501 @@
+                Sequence::parse_space,
+                Sequence::parse_upper,
+                Sequence::parse_xdigit,
+                Sequence::parse_char_equal,
+                // NOTE: This must be the last one
+                Sequence::parse_char,
+            )),
+        )))(input)
+        .map(|(_, r)| r)
+        .unwrap()
+    }
+
+    pub fn dissolve(self) -> Box<dyn Iterator<Item = char>> {
+        match self {
+            Sequence::Char(c) => Box::new(std::iter::once(c)),
+            Sequence::CharRange(r) => r,
+            Sequence::CharStar(c) => Box::new(std::iter::repeat(c)),
+        }
+    }
+
+    /// Sequence parsers
+
+    fn parse_char(input: &str) -> IResult<&str, Sequence> {
+        anychar(input).map(|(l, r)| (l, Sequence::Char(r)))
+    }
+
+    fn parse_backslash(input: &str) -> IResult<&str, Sequence> {
+        preceded(tag("\\"), anychar)(input).map(|(l, a)| {
+            let c = match a {
+                'a' => Sequence::Char(unicode_table::BEL),
+                'b' => Sequence::Char(unicode_table::BS),
+                'f' => Sequence::Char(unicode_table::FF),
+                'n' => Sequence::Char(unicode_table::LF),
+                'r' => Sequence::Char(unicode_table::CR),
+                't' => Sequence::Char(unicode_table::HT),
+                'v' => Sequence::Char(unicode_table::VT),
+                x => Sequence::Char(x),
+            };
+            (l, c)
+        })
+    }
+
+    fn parse_octal(input: &str) -> IResult<&str, Sequence> {
+        map_opt(
+            preceded(tag("\\"), recognize(many_m_n(1, 3, one_of("01234567")))),
+            |out: &str| {
+                u32::from_str_radix(out, 8)
+                    .map(|u| Sequence::Char(char::from_u32(u).unwrap()))
+                    .ok()
+            },
+        )(input)
+    }
+
+    fn parse_char_range(input: &str) -> IResult<&str, Sequence> {
+        separated_pair(anychar, tag("-"), anychar)(input).map(|(l, (a, b))| {
+            (l, {
+                let (start, end) = (u32::from(a), u32::from(b));
+                Sequence::CharRange(Box::new((start..=end).filter_map(std::char::from_u32)))
+            })
+        })
+    }
+
+    fn parse_char_star(input: &str) -> IResult<&str, Sequence> {
+        tuple((tag("["), anychar, tag("*]")))(input)
+            .map(|(l, (_, c, _))| (l, Sequence::CharStar(c)))
+    }
+
+    fn parse_char_repeat(input: &str) -> IResult<&str, Sequence> {
+        tuple((
+            tag("["),
+            anychar,
+            tag("*"),
+            // TODO: Extend this to support octal as well. Octal starts with 0.
+            take_while1(|c: char| c.is_digit(10)),
+            tag("]"),
+        ))(input)
+        .map(|(l, (_, c, _, n, _))| {
+            (
+                l,
+                Sequence::CharRange(Box::new(std::iter::repeat(c).take(n.parse().unwrap()))),
+            )
+        })
+    }
+
+    fn parse_alnum(input: &str) -> IResult<&str, Sequence> {
```

Comment:
These functions for `[:something:]` could still use `value` to make it a little bit cleaner.

Reply:
I can't use it because it requires the `Sequence` to be cloned. But I can't do it because its a Box.

Reply:
Ah interesting! Then it's alright of course